sharding-core/sharding-core-common/src/test/java/org/apache/shardingsphere/core/metadata/table/TableMetasTest.java

```diff
@@ -0,0 +1,87 @@
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.core.metadata.table;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import org.apache.shardingsphere.core.metadata.column.ColumnMetaData;
+import org.junit.Test;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+
+public final class TableMetasTest {
+    
+    @Test
+    public void assertGetOk() {
+        TableMetaData tableMetaData = mock(TableMetaData.class);
+        TableMetas tableMetas = new TableMetas(ImmutableMap.of("tableMetaData", tableMetaData));
+        assertThat(tableMetas.get("tableMetaData"), is(tableMetaData));
+    }
+    
+    @Test
+    public void assertPutOk() {
+        TableMetas tableMetas = new TableMetas(Collections.<String, TableMetaData>emptyMap());
+        TableMetaData tableMetaData = mock(TableMetaData.class);
+        tableMetas.put("tableMetaData", tableMetaData);
+        assertThat(tableMetas.get("tableMetaData"), is(tableMetaData));
+    }
+    
+    @Test
+    public void assertRemoveOk() {
+        TableMetas tableMetas = new TableMetas(ImmutableMap.of("tableMetaData", mock(TableMetaData.class)));
+        tableMetas.remove("tableMetaData");
+        assertNull(tableMetas.get("tableMetaData"));
+    }
+    
+    @Test
+    public void assertContainsTableOk() {
+        assertTrue(new TableMetas(ImmutableMap.of("tableMetaData", mock(TableMetaData.class))).containsTable("tableMetaData"));
+    }
+    
+    @Test
+    public void assertContainsColumnOk() {
+        TableMetaData tableMetaData = new TableMetaData(Collections.singletonList(new ColumnMetaData("name", "dataType", false)), Collections.<String>emptyList());
+        assertTrue(new TableMetas(ImmutableMap.of("tableMetaData", tableMetaData)).containsColumn("tableMetaData", "name"));
+    }
+    
+    @Test
+    public void assertGetAllColumnNamesOkWhenContainsKey() {
+        TableMetaData tableMetaData = new TableMetaData(Collections.singletonList(new ColumnMetaData("name", "dataType", false)), Collections.<String>emptyList());
+        assertThat(new TableMetas(ImmutableMap.of("tableMetaData", tableMetaData)).getAllColumnNames("tableMetaData"), is((List<String>) Lists.newArrayList("name")));
```

Comment:
Can use `Collections.singletionList()` to instead of `Lists.newArrayList()` if list have single element only.